resources/lib/UnityLDAP.php

```diff
@@ -115,67 +115,58 @@ public function getDefUserShell()
   //
   // ID Number selection functions
   //
-    public function getNextUIDNumber()
+    public function getNextUIDNumber($UnitySQL)
     {
-        $users = $this->userOU->getChildrenArray(true);
+        $max_uid = $UnitySQL->getSiteVar('MAX_UID');
+        $new_uid = $max_uid + 1;
 
-      // This could become inefficient with more users
-        usort($users, function ($a, $b) {
-            return $a["uidnumber"] <=> $b["uidnumber"];
-        });
-
-        $id = self::ID_MAP[0];
-        foreach ($users as $acc) {
-            if ($id == $acc["uidnumber"][0]) {
-                $id++;
-            } else {
-                if (!$this->GIDNumInUse($id)) {
-                    break;
-                }
+        $is_already_in_use = $this->UIDNumInUse($new_uid);
+
+        if ($is_already_in_use) {
+            while ($this->UIDNumInUse($new_uid)) {
+                $new_uid++;
             }
         }
 
-        return $id;
+        $UnitySQL->updateSiteVar('MAX_UID', $new_uid);
+
+        return $new_uid;
     }
 
-    public function getNextPiGIDNumber()
+    public function getNextPiGIDNumber($UnitySQL)
     {
-        $groups = $this->pi_groupOU->getChildrenArray(true);
-
-        usort($groups, function ($a, $b) {
-            return $a["gidnumber"] <=> $b["gidnumber"];
-        });
-
-        $id = self::PI_ID_MAP[0];
-        foreach ($groups as $acc) {
-            if ($id == $acc["gidnumber"][0]) {
-                $id++;
-            } else {
-                break;
+        $max_pigid = $UnitySQL->getSiteVar('MAX_PIGID');
+        $new_pigid = $max_pigid + 1;
+
+        $pi_groups = $this->pi_groupOU->getChildrenArray(true);
+
+        foreach ($pi_groups as $group) {
+            if ($group["gidnumber"][0] == $new_pigid) {
+                $new_pigid++;
             }
         }
 
-        return $id;
+        $UnitySQL->updateSiteVar('MAX_PIGID', $new_pigid);
+
+        return $new_pigid;
     }
 
-    public function getNextOrgGIDNumber()
+    public function getNextOrgGIDNumber($UnitySQL)
     {
-        $groups = $this->org_groupOU->getChildrenArray(true);
+        $max_gid = $UnitySQL->getSiteVar('MAX_GID');
+        $new_gid = $max_gid + 1;
 
-        usort($groups, function ($a, $b) {
-            return $a["gidnumber"] <=> $b["gidnumber"];
-        });
+        $groups = $this->org_groupOU->getChildrenArray(true);
 
-        $id = self::ORG_ID_MAP[0];
-        foreach ($groups as $acc) {
-            if ($id == $acc["gidnumber"][0]) {
-                $id++;
-            } else {
-                break;
+        foreach ($groups as $group) {
+            if ($group["gidnumber"][0] == $new_gid) {
+                $new_gid++;
             }
         }
 
-        return $id;
+        $UnitySQL->updateSiteVar('MAX_GID', $new_gid);
+
+        return $new_gid;
     }
 
     private function UIDNumInUse($id)
```

Comment:
I'd prefer if you made 2 additional methods for `GIDNumInUse` and `PIGIDNumInUse` to keep it consistent.